tensorflow/core/kernels/data/zip_dataset_op.cc

```diff
@@ -140,22 +140,20 @@ class ZipDatasetOp::Dataset : public DatasetBase {
       *end_of_sequence = false;
       for (const auto& input_impl : input_impls_) {
         std::vector<Tensor> input_tensors;
-        bool end_of_sequence_component = false;
-        Status status_component = input_impl->GetNext(ctx, &input_tensors, &end_of_sequence_component);
-        if (end_of_sequence_component) {
-          *end_of_sequence = end_of_sequence_component;
+        bool component_end_of_sequence = false;
+        Status component_status = input_impl->GetNext(ctx, &input_tensors, &component_end_of_sequence);
+        *end_of_sequence |= component_end_of_sequence;
+        if (!component_status.ok() && status.ok()) {
+          status.Update(component_status);
         }
```

Comment:
You don't need the if statement. `status.Update(component_status)` only updates `status` if it is OK.

Reply:
@jsimsa Thanks! Didn't know this usage before. Updated.